File: app/src/main/java/com/greenwald/aaron/ridetracker/model/SegmentPoint.java
package com.greenwald.aaron.ridetracker.model;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.Gson;

import java.util.Date;

public class SegmentPoint {
    public double latitude;
    public double longitude;
    public double accuracy;
    public Date dateTime;
    public double altitude;

    public SegmentPoint(double latitude, double longitude, double accuracy, Date dateTime) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.accuracy = accuracy;
        this.dateTime = dateTime;
    }

    public static SegmentPoint fromString(String string) {
        Gson gson = new Gson();
        return gson.fromJson(string, SegmentPoint.class);
    }

    public String toString() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }

    public LatLng getLatLng() {
        return new LatLng(this.latitude, this.longitude);
    }
}

File: app/src/main/java/com/greenwald/aaron/ridetracker/model/SegmentPoint.kt
package com.greenwald.aaron.ridetracker.model

import com.google.android.gms.maps.model.LatLng
import com.google.gson.Gson

import java.util.Date

data class SegmentPoint(val latitude: Double,
                        val longitude: Double,
                        val accuracy: Double,
                        val dateTime: Date,
                        val altitude: Meters,
                        val altitudeChange: Meters = Meters(0.0),
                        val elapsedTime: Milliseconds = Milliseconds(0),
                        val distance: Meters = Meters(0.0)) {

    val latLng: LatLng
        get() = LatLng(this.latitude, this.longitude)

    override fun toString(): String {
        val gson = Gson()
        return gson.toJson(this)
    }

    companion object {
        fun fromString(string: String): SegmentPoint {
            val gson = Gson()
            return gson.fromJson(string, SegmentPoint::class.java)
        }
    }
}
